packages/react-instantsearch-core/src/connectors/connectVoiceSearch.js

```diff
@@ -0,0 +1,156 @@
+import PropTypes from 'prop-types';
+import createConnector from '../core/createConnector';
+import {
+  cleanUpValue,
+  refineValue,
+  getCurrentRefinementValue,
+  getIndexId,
+} from '../core/indexUtils';
+
+function getId() {
+  return 'query';
+}
+
+function getAdditionalId() {
+  return 'additionalVoiceParameters';
+}
+
+function getCurrentRefinementQuery(props, searchState, context) {
+  const id = getId();
+  const currentRefinement = getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    id,
+    ''
+  );
+
+  if (currentRefinement) {
+    return currentRefinement;
+  }
+  return '';
+}
+
+function getCurrentRefinementAdditional(props, searchState, context) {
+  const id = getAdditionalId();
+  const currentRefinement = getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    id,
+    ''
+  );
+
+  if (currentRefinement) {
+    return currentRefinement;
+  }
+  return {};
+}
+
+function refine(props, searchState, nextRefinement, context) {
+  const id = getId();
+  const voiceParams = getAdditionalId();
+
+  const nextValue = {
+    [id]: nextRefinement,
+    [voiceParams]:
+      typeof props.additionalQueryParameters === 'function'
+        ? {
+            queryLanguages: props.language
```

Comment:
queryLanguages should be set based on language, the other ones when you pass the argument. We changed this in InstantSearch.js